homeassistant/components/mqtt/discovery.py

```diff
@@ -163,11 +163,9 @@ class MQTTConfig(dict):
                 and component in IMPLICIT_STATE_TOPIC_COMPONENTS
             ):
                 # state_topic not specified, infer from discovery topic
-                payload[CONF_STATE_TOPIC] = "{}/{}/{}{}/state".format(
-                    discovery_topic,
-                    component,
-                    "%s/" % node_id if node_id else "",
-                    object_id,
+                payload[CONF_STATE_TOPIC] = (
+                    f"{discovery_topic}/{component}/"
+                    f"{f'{node_id}/' if node_id else ''}{object_id}/state"
```

Comment:
This will be easier to read if we save the formatted node_id in a variable first.
```py
fmt_node_id = f"{node_id}/" if node_id else ""
payload[CONF_STATE_TOPIC] = (
    f"{discovery_topic}/{component}/"
    f"{fmt_node_id}{object_id}/state"
)

Reply:
```suggestion
                    fmt_node_id = f"{node_id}/" if node_id else ""
                    payload[CONF_STATE_TOPIC] = (
                        f"{discovery_topic}/{component}/"
                        f"{fmt_node_id}{object_id}/state"
                    )
```